Tests/ApolloCodegenTests/CodeGenAST/ASTSelectionSetTests.swift

```diff
@@ -0,0 +1,2415 @@
+
+    let selectionSet = CompilationResult.SelectionSet.mock(
+      selections: [
+        .field(.mock(
+          "A",
+          type: .named(Object_A),
+          selectionSet: .mock(
+            parentType: Object_A,
+            selections: [
+              .field(.mock("B", type: .named(GraphQLScalarType.integer())))
+            ]
+          ))),
+        .field(.mock(
+          "A",
+          type: .named(Object_A),
+          selectionSet: .mock(
+            parentType: Object_A,
+            selections: [
+              .field(.mock("C", type: .named(GraphQLScalarType.integer())))
+            ]
+          )))
+      ]
+    )
+
+    let expected: [CompilationResult.Selection] = [
+      .field(.mock(
+        "A",
+        type: .named(Object_A),
+        selectionSet: .mock(
+          parentType: Object_A,
+          selections: [
+            .field(.mock("B", type: .named(GraphQLScalarType.integer()))),
+            .field(.mock("C", type: .named(GraphQLScalarType.integer())))
+          ]
+        )))
+    ]
+
+    // when
+    let actual = ASTSelectionSet(selectionSet: selectionSet, compilationResult: mockCompilationResult)
+
+    // then
+    expect(actual.selections).to(equal(expected))
+  }
+
+  /// Example:
+  /// query {
+  ///   A {
+  ///     B
+  ///     C
+  ///   }
+  ///   A {
+  ///     B
+  ///     D
+  ///   }
+  /// }
+  ///
+  /// Expected:
+  /// Query.Selections: {
+  ///   A {
+  ///     B
+  ///     C
+  ///     D
+  ///   }
+  /// }
+  func test__selections__givenFieldSelectionsWithSameResponseKey_onObjectWithSameAndDifferentChildSelections_mergesChildSelectionsAndDoesNotDuplicateFields() {
+    // given
+    let Object_A = GraphQLObjectType.mock("A")
+
+    let selectionSet = CompilationResult.SelectionSet.mock(
+      selections: [
+        .field(.mock(
+          "A",
+          type: .named(Object_A),
+          selectionSet: .mock(
+            parentType: Object_A,
+            selections: [
+              .field(.mock("B", type: .named(GraphQLScalarType.integer()))),
+              .field(.mock("C", type: .named(GraphQLScalarType.integer()))),
+            ]
+          ))),
+        .field(.mock(
+          "A",
+          type: .named(Object_A),
+          selectionSet: .mock(
+            parentType: Object_A,
+            selections: [
+              .field(.mock("B", type: .named(GraphQLScalarType.integer()))),
+              .field(.mock("D", type: .named(GraphQLScalarType.integer()))),
+            ]
+          )))
+      ]
+    )
+
+    let expected: [CompilationResult.Selection] = [
+      .field(.mock(
+        "A",
+        type: .named(Object_A),
+        selectionSet: .mock(
+          parentType: Object_A,
+          selections: [
+            .field(.mock("B", type: .named(GraphQLScalarType.integer()))),
+            .field(.mock("C", type: .named(GraphQLScalarType.integer()))),
+            .field(.mock("D", type: .named(GraphQLScalarType.integer()))),
```

Comment:
Is there a way to set up `.named` as a default value (since that's not really what you're testing here)?

Reply:
Done! Thanks for the suggestion!